### DISCUSSION TITLE
HP system issues and overhaul proposal

### DISCUSSION THREAD

**Discussion by mk56-spn** [ppy/osu#14454] [Ideas] [answered]
As I brought up the other day in the osu!lazer channel in the Dev server on Discord (and I am readressing here since it got buried and went unanswered, also atlas spaceman told me it would be more appropriate to post here)  I feel lazer is in bad need of a HP overhaul, here is why:

1. Upon the removal of classics note-lock system which stream maps seemed to heavily depend upon for pass difficulty, the maps have become ridiculously easy to pass (first part of  https://www.youtube.com/watch?v=HvJ5-nRoBGQ) due to the fact that you can now miss anywhere in the stream and continue hitting notes after immediately  and due to the sheer volume of notes  in streams  compared to jumps  you can hit half of the notes regardless of accuracy and still survive on most stream maps  ( honesty ( dishonest) ( bibbity bill) , raise my sword (sotarks) ,etc...) this allows you to pass 7-8 star HP 5  stream maps with 40% accuracy, in my opinion completely devaluing any feeling of  achievement you would have gotten previously.

2. On the other end of the scale is jump maps which seem to range from mildly harder than their same star stream map equivalents to pass, to HP failing you with no more than 3  non sequential misses , even on the same HP ( i tested on 5 but it applies  to all the HP range ) proof of this is 1:01 onwards in this video (https://www.youtube.com/watch?v=HvJ5-nRoBGQ) where the 6,99 star difficulty  of the loved sakura no uta mapset (https://osu.ppy.sh/beatmapsets/811908#osu/1962676) sends my HP bar to nearly zero at 94 acc and 2 or so misses whilst the  almost 8 star difficulty shown at 1:41( https://www.youtube.com/watch?v=HvJ5-nRoBGQ)  allows me to miss an entire 7 slider cross screen jump section before finally HP failing me, this is especially concerning since they are both HP5 as well and none have any particularly weird sliders that would cause HP drop from slider end drops.

I am well aware that these may be outliers overall but they are still far too common to ignore entirely 

Proposed solution: 

I would like to rework the hp system to account for object density better by giving lower hp refill per circle / slider when there is a large amount of them in close proximity on the timeline, and  to address the imbalance among same HP jump maps 

I am aware that this may not seem like a priority since a large part of the community does not care for passes but i think its of utmost importance that a solid HP system is in place by the time  any sort of permanent and visible score submission goes live since after that changing HP would create problematic discrepancies like, for example, the same play HP failing you or  letting you pass depending on when it was done, or to avoid that scores would have to be recalculated and removed if new hp system updates deemed them no longer to be passes ( or potentially given NF status when that happens )

**Comment by smoogipoo** [comment, marked as answer]
It's not just it being too easy and jump maps being too hard. For example: https://github.com/ppy/osu/issues/7975#issuecomment-898874189 - even more normal maps can be hard to pass.

Density may be a consideration, but I also don't want to make the system complex. Any layman should be able to understand how HP works - currently it's described as the minimum percentage with a perfect play, and I imagine with density it should be able to be described as something like "percent of HP drained in the period of two hitobjects spaced one beat apart".

**Reply by mk56-spn** [reply]
I understand where you are coming from and you make a good point that my bandaid suggestion of factoring in more metrics to balance out the pass difficulty across maps might do more harm than good.
